### PR TITLE
Add a report flag if generated by the OEM

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-report.h [hunk 1]
```diff
@@ -25,6 +25,41 @@ struct _FwupdReportClass {
...
+ * The report was generated by the OEM.
+ *
+ * Since: 1.9.1
+ */
+#define FWUPD_REPORT_FLAG_FROM_OEM (1llu << 0)
+
+/**
+ * FWUPD_REPORT_FLAG_UNKNOWN:
+ *
+ * The plugin flag is unknown.
```
Comment (superm1) [inline libfwupd/fwupd-report.h:49]: Report

File: libfwupd/fwupd-report.h [hunk 2]
```diff
@@ -25,6 +25,41 @@ struct _FwupdReportClass {
...
+ * This is usually caused by a mismatched libfwupdplugin and daemon.
+ *
+ * Since: 1.9.1
+ */
+#define FWUPD_REPORT_FLAG_UNKNOWN G_MAXUINT64
+
+/**
+ * FwupdReportFlags:
+ *
```
Comment (superm1) [inline libfwupd/fwupd-report.h:58]: Report

File: src/fu-release.c [hunk 3]
```diff
@@ -323,6 +323,10 @@ fu_release_load_test_result(FuRelease *self, XbNode *n, GError **error)
 	if (custom != NULL) {
 		for (guint i = 0; i < custom->len; i++) {
 			XbNode *c = g_ptr_array_index(custom, i);
+			if (g_strcmp0(xb_node_get_attr(c, "key"), "FromOEM") == 0) {
```
Comment (superm1) [inline src/fu-release.c:326]: Isn't this named improperly? It's more like from a trusted source. For example if Dell shipped a dock update but it was validated by say Google in ChromeOS and Google happened to push a signed report it should be given some weight just like this I would think.

File: libfwupd/fwupd-report.h (hunk 1 above)
Comment (hughsie) [reply libfwupd/fwupd-report.h:49]: Fixed, ta.

File: libfwupd/fwupd-report.h (hunk 2 above)
Comment (hughsie) [reply libfwupd/fwupd-report.h:58]: Fixed.

File: src/fu-release.c (hunk 3 above)
Comment (hughsie) [reply src/fu-release.c:326]: The logic I was aiming for was to have something like `DistroId:chromeos,VendorId:$SELF;DistroId:chromeos,VendorId:19` i.e. a trusted report is anything by the original OEM, or by the Google QA team.

**Review by superm1 — APPROVED** [review]

**Pull request merged**